cpp/ql/lib/semmle/code/cpp/exprs/Expr.qll

```diff
@@ -534,6 +534,9 @@ class Expr extends StmtParent, @expr {
 
   /** Gets the function containing this control-flow node. */
   override Function getControlFlowScope() { result = this.getEnclosingFunction() }
+
+  /** Holds if this expression could be the return value of an implicitly declared function. */
+  predicate mayBeFromImplicitlyDeclaredFunction() { none() }
```

Comment:
Why not just make this a predicate of `Call`?

Reply:
Because we need to ask an `Expr` if it came from an implicitly-declared function. I suppose, we could have done `arg.(Call).mayBeFromImplicitlyDeclaredFunction()`?

Reply:
Yeah, I think that would be better.